app/javascript/controllers/tooltip_controller.js

```diff
@@ -0,0 +1,82 @@
+import { Controller } from '@hotwired/stimulus'
+import {
+  computePosition,
+  flip,
+  shift,
+  offset,
+  arrow
+} from '@floating-ui/dom';
+
+export default class extends Controller {
+  static targets = ["element", "tooltip"];
+  static values = {
+    placement: String,
+    offset: Number,
+    crossAxis: Number
+  };
+
+  initialize() {
+    this.placementOption = this.placementValue || "top"
+    this.offsetOption = this.offsetValue || 10
+    this.crossAxisOption = this.crossAxisValue || 0
+    this.alignmentAxisOption = this.alignmentValue || null
+    this.arrowElement = document.querySelector('#arrow');
+  }
```

Comment:
With Stimulus we can provide default values for the value targets, which would allow us to remove this method and replace the option variables with the targets.

```js
static values = { 
  placement: { type: String, default: "top" },
  offset: { type: Number, default: 10 },
  crossAxis: { type: Number, default: 0 },
}
```

Reply:
[03344a2](https://github.com/maybe-finance/maybe/pull/1065/commits/03344a2677142d7621bd29999620f0a058420035)